src/OrleansCodeGenerator/GrainMethodInvokerGenerator.cs

```diff
@@ -255,7 +255,7 @@ private static MethodDeclarationSyntax GenerateInvokeMethod(Type grainType, Meth
             for (var i = 0; i < methodParameters.Count; i++)
             {
                 var parameter = methodParameters[i];
-                var parameterType = parameter.ParameterType.GetTypeSyntax();
+                var parameterType = parameter.ParameterType.GetTypeSyntaxAsDynamic();
```

Comment:
Yeah, I wasn't even sure it should be a separate function. But I wasn't sure what the side effects would be of me modifying GetTypeSyntax() to handle both methods, or going upstream to the Orleans.Runtime, or if that is even the right more.